thoth-errors/src/database_errors.rs

```diff
@@ -0,0 +1,268 @@
+use phf::phf_map;
+use phf::Map;
+
+use crate::ThothError;
+
+/// A map of database constraint name and a corresponding error to output
+/// when the constraint is violated.
+///
+/// To obtain a list of unique and check constraints:
+/// ```sql
+/// SELECT conname
+/// FROM pg_catalog.pg_constraint con
+/// INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+/// WHERE nsp.nspname = 'public'
+/// AND contype in ('u', 'c');
+/// ```
+static DATABASE_CONSTRAINT_ERRORS: Map<&'static str, &'static str> = phf_map! {
+    "publisher_uniq_idx" => "A publisher with this name already exists.",
+    "imprint_uniq_idx" => "An imprint with this name already exists.",
+    "doi_uniq_idx" => "A work with this DOI already exists.",
+    "language_uniq_work_idx" => "Duplicate language code.",
+    "series_issn_print_idx" => "A series with this ISSN already exists.",
+    "series_issn_digital_idx" => "A series with this ISSN already exists.",
```

Comment:
Specify "print ISSN"/"digital ISSN" as the page where they'll be displayed has two ISSN fields.